pkg/controller/servicebindingrequest/nested/acc/acc.go

```diff
@@ -0,0 +1,46 @@
+package acc
+
+import (
+	"errors"
+
+	"github.com/imdario/mergo"
+)
+
+const valuesKey = "values"
+
+// Acc is a value accumulator.
+type Acc map[string]interface{}
+
+// UnsupportedTypeErr is returned when an unsupported type is encountered.
+var UnsupportedTypeErr = errors.New("unsupported type")
+
+// Accumulate accumulates the `val` value. An error is returned in the case
+// `val` contains an unsupported type.
+func (a Acc) Accumulate(val interface{}) error {
```

Comment:
Just to check if I understand correctly: since all values are stored under the same key, this method actually implements a sort of set addition?

Reply:
Yes. It prepares a data structure for `mergo.Merge` to merge with the existing, collected data.

Reply:
Can we then call the structure `Set`? Instead of `Acc`? And this function could become a simple `Add`?

Reply:
I'm not sure a Set would be the right term, since all leaves will be subject to be merged.